src/ui/app/app.jsx

```diff
@@ -0,0 +1,77 @@
+/*
+Copyright (c) 2021 - present / Neuralmagic, Inc. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+   http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing,
+software distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+import React, { useEffect } from "react";
+import { Route } from "react-router-dom";
+import { useDispatch, useSelector } from "react-redux";
+import { ThemeProvider } from "@material-ui/core/styles";
+import Cookies from "universal-cookie";
+import moment from "moment";
+
+import makeTheme, { useDarkMode } from "./app-theme";
+import { authThunk, selectAuthState, setAuthToken } from "../store";
+import { makeContentRoutes } from "./../routes";
+import makeStyles from "./app-styles";
+
+function App() {
+  const useStyles = makeStyles();
+  const classes = useStyles();
+  const dispatch = useDispatch();
+
+  const darkMode = useDarkMode()[0];
+  const theme = makeTheme(darkMode);
+
+  const authState = useSelector(selectAuthState);
+
+  useEffect(() => {
+    if (authState.status === "idle") {
+      const cookies = new Cookies();
+      const authToken = cookies.get("NM_AUTH_TOKEN");
+      if (authToken) {
+        dispatch(setAuthToken(authToken));
+      } else {
+        dispatch(authThunk());
+      }
+    } else if (authState.status === "succeeded") {
+      const cookies = new Cookies();
+
+      const authToken = cookies.get("NM_AUTH_TOKEN");
+      if (authToken !== authState.token) {
+        const expires = moment().add(7, "days").toDate();
```

Comment:
can we get a lookup from the API for when the token is supposed to expire? This could cause us some issues in the future if we ever decide to change it